theano/tensor/opt.py

```diff
@@ -6251,8 +6250,8 @@ def local_mul_exp(node):
         return False
 
 compile.optdb.register('local_log_sum_exp',
-                       in2out([local_log_sum_exp,
-                               local_mul_exp],
+                       in2out(local_log_sum_exp,
+                              local_mul_exp,
                               ignore_newtrees=True),
                        # between stabilize at 1.5 and specialize at 2
                        # local_mul_exp should be after local_greedy_distributor
```

Comment:
Thanks for the hint. With your patch, `x**y* y**(-y)` gets optimized to 
`Elemwise{Pow}[(0, 1)] [id A] ''   
 |<TensorType(float64, vector)> [id B]
 |Elemwise{Add}[(0, 1)] [id C] ''   
   |<TensorType(float64, vector)> [id D]
   |Elemwise{neg,no_inplace} [id E] ''   
     |<TensorType(float64, vector)> [id D]
`

Something is still missing :)